.github/workflows/static-checks.yml

```diff
@@ -56,6 +56,12 @@ jobs:
         run: |
           echo "JAVA_HOME=$JAVA_HOME_${{ env.java_version }}_X64" >> $GITHUB_ENV
 
+      - name: Cache Maven packages
```

Comment:
Because we have a `matrix` of `java`, should we include `${{ matrix.java }}` in the cache key? I'm not sure if the `java` versions matters for what `mvn` does, but, it might…?

I also wonder if we can use `actions/setup-java` with `cache: 'maven'`, which according to documentation (https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#about-caching-workflow-dependencies) automatically configures caching with the appropriate variables.